habitat/datasets/object_nav/object_nav_dataset.py

```diff
@@ -26,13 +30,58 @@ class ObjectNavDatasetV1(PointNavDatasetV1):
     """
     category_to_task_category_id: Dict[str, int]
     category_to_scene_annotation_category_id: Dict[str, int]
-    episodes: List[NavigationEpisode]
+    episodes: List[ObjectGoalNavEpisode]
     content_scenes_path: str = "{data_path}/content/{scene}.json.gz"
+    goals_by_category: Dict[str, List[ObjectGoal]]
+
+    @staticmethod
+    def dedup_goals(dset: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
you're doing change in place on `dset`, no need to return it.

Reply:
The change in-place is done on purpose.  It makes a considerable difference on the time it takes to construct the list of episodes if you do this change in-place first.